### PR TITLE
Extract createLoadIfStore logic from the general helper to the caller

### PR DESCRIPTION
**Pull request by rmnattas**

### CODE REVIEW

**Comment by VermaSh** [comment]
Can you please comment the failure this fixes

File: runtime/compiler/optimizer/J9TransformUtil.cpp [hunk 1]
```diff
@@ -155,30 +155,31 @@ TR::Node *
 J9::TransformUtil::generateArrayOffsetTrees(TR::Compilation *comp, TR::Node *indexNode, TR::Node *elementSizeNode, int32_t elementSize, bool useShiftOpCode)
    {
    TR::Node *offsetNode = indexNode->createLongIfNeeded();
+   if (elementSizeNode)
+      elementSizeNode = elementSizeNode->createLongIfNeeded();
+   // TODO_sverma: Need to investigate why using convertStoreDirectToLoadWithI2LIfNeeded
```
Comment (VermaSh) [inline runtime/compiler/optimizer/J9TransformUtil.cpp]: We can probably delete these lines

File: runtime/compiler/optimizer/J9TransformUtil.cpp [hunk 2]
```diff
@@ -155,30 +155,31 @@ TR::Node *
 J9::TransformUtil::generateArrayOffsetTrees(TR::Compilation *comp, TR::Node *indexNode, TR::Node *elementSizeNode, int32_t elementSize, bool useShiftOpCode)
    {
    TR::Node *offsetNode = indexNode->createLongIfNeeded();
+   if (elementSizeNode)
+      elementSizeNode = elementSizeNode->createLongIfNeeded();
```
Comment (VermaSh) [inline runtime/compiler/optimizer/J9TransformUtil.cpp]: Let's create a local variable for `elementSizeNode` for better readability

**Review by VermaSh — CHANGES_REQUESTED** [review]

**Comment by rmnattas** [comment]
> Can you please comment the failure this fixes

Directly address failure in `jdk_imageio_0` which usually appears in P & Z, of `Failed. Execution failed: 'main' threw exception: java.lang.IndexOutOfBoundsException`.

Might solve other failures too.

File: runtime/compiler/optimizer/J9TransformUtil.cpp (hunk 1 above)
Comment (rmnattas) [reply runtime/compiler/optimizer/J9TransformUtil.cpp]: Comment lines was coming from a merge conflict but can be removed

File: runtime/compiler/optimizer/J9TransformUtil.cpp (hunk 2 above)
Comment (rmnattas) [reply runtime/compiler/optimizer/J9TransformUtil.cpp]: Not sure if something like this is better... you still have to check for null-ness, and I can't think of a better name than `elementSizeNode`, anything in mind?

```
   TR::Node *strideNode = elementSizeNode;
   if (strideNode)
      strideNode = strideNode->createLongIfNeeded();
```

File: runtime/compiler/optimizer/J9TransformUtil.cpp (hunk 1 above)
Comment (VermaSh) [reply runtime/compiler/optimizer/J9TransformUtil.cpp]: We aren't calling that function anymore so no need to keep the comment around.

File: runtime/compiler/optimizer/J9TransformUtil.cpp (hunk 2 above)
Comment (VermaSh) [reply runtime/compiler/optimizer/J9TransformUtil.cpp]: yea, I think stride is a good name for the variable.

**Comment by rmnattas** [comment]
Done with changes

File: runtime/compiler/optimizer/IdiomRecognitionUtils.cpp [hunk 3]
```diff
@@ -892,16 +892,17 @@ createArrayAddressTree(TR::Compilation *comp, bool is64bit, TR::Node *baseNode,
...
+   TR::Node *aload = createLoad(baseNode);
    if (indexNode->getOpCodeValue() == TR::iconst && indexNode->getInt() == 0)
       {
-      top = TR::TransformUtil::generateArrayStartTrees(comp, baseNode);
+      top = TR::TransformUtil::generateArrayStartTrees(comp, aload);
       }
    else
       {
       // c2 = createBytesFromElement(comp, is64bit, indexNode, multiply);
       // createBytesFromElement esentially creates offset trees
```
Comment (VermaSh) [inline runtime/compiler/optimizer/IdiomRecognitionUtils.cpp]: We don't need to keep these comment lines around. Can you please remove those as well.

**Review by VermaSh — CHANGES_REQUESTED** [review]
Looks good to me, just one small suggestion

**Pull request merged**